### PR TITLE
Improved UNet Lesion Segmentation

### PR DESCRIPTION
**Pull request by georgelaugher**
_45807152 LAUGHER, George_

**Full solution description can be found in README.md**

This PR contains a solution to the ISIC Lesion segmentation problem (_Task 1_) utilizing an Improved UNet model framework.
The recognition solution is implemented using TensorFlow (_2.9.1_) and separated into the following files:
* **modules.py** - contains Improved UNet model class and components.
* **dataset.py** - provides a data loader for the 2017 ISIC data.
* **train.py** - used to train the Improved UNet model on the 2017 ISIC data.
* **predict.py** - contains functions to predict segmentation mask from a trained model.
* **utility.py** - contains metric calculations used when training, validating and testing the Improved UNet model.
* **config.py** - provides the ability to adjust training and input parameters.

Additionally, there are example images and prediction comparisons contained within the _figures_ subfolder.

Overall, the model performed very well and achieved a dice coefficient of _**0.9528**_ and IoU of _**0.9104**_ on 'unseen' testing data.

Happy to implement any suggestions you may have. Hope to see this solution accepted into the repository soon!

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
**This is an initial inspection, no action is required at this point**
- Code: OK
- Results: OK
- Readme: OK
- Commit Messages: OK
- Other comments: N/A

**Comment by shakes76** [comment]
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate use and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage 
Module present
Commenting
No Data leakage
Difficulty: Easy -10
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
